scout/adapter/mongo/panel.py

```diff
@@ -462,89 +463,53 @@ def reset_pending(self, panel_obj):
         )
         return updated_panel
 
-    def apply_pending(self, panel_obj, version):
+    def apply_pending(self, panel_obj: dict, version: float) -> str:
         """Apply the pending changes to an existing gene panel or create a new version of the same panel.
 
-        Args:
-            panel_obj(dict): panel in database to update
-            version(double): panel version to update
-
         Returns:
             inserted_id(str): id of updated panel or the new one
         """
 
-        updates = {}
         new_panel = deepcopy(panel_obj)
         new_panel["pending"] = []
         new_panel["date"] = dt.datetime.now()
-        info_fields = [
-            "disease_associated_transcripts",
-            "inheritance_models",
-            "custom_inheritance_models",
-            "reduced_penetrance",
-            "mosaicism",
-            "database_entry_version",
-            "comment",
-        ]
         new_genes = []
 
-        for update in panel_obj.get("pending", []):
-            hgnc_id = update["hgnc_id"]
-
-            # If action is add we create a new gene object
-            if update["action"] != "add":
-                updates[hgnc_id] = update
-                continue
-            info = update.get("info", {})
-            gene_obj = {"hgnc_id": hgnc_id, "symbol": update["symbol"]}
-
-            for field in info_fields:
-                if field in info:
-                    gene_obj[field] = info[field]
-            new_genes.append(gene_obj)
+        # Process 'add' actions and gather updates
+        updates = {u["hgnc_id"]: u for u in panel_obj.get("pending", []) if u["action"] != "add"}
```

Comment:
This is quite compact, but looks good to me. Let's hope we don't need to change it in a few months.